File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length;
        if (n <= 2) return n;

        int minIdx = 0;
        int maxIdx = 0;

        // Find indices of min and max elements
        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[minIdx]) minIdx = i;
            if (nums[i] > nums[maxIdx]) maxIdx = i;
        }

        int left = Math.min(minIdx, maxIdx);
        int right = Math.max(minIdx, maxIdx);

        // Scenario 1: Both from front
        int costFront = right + 1;

        // Scenario 2: Both from back
        int costBack = n - left;

        // Scenario 3: One from front, one from back
        int costBoth = (left + 1) + (n - right);

        return Math.min(costFront, Math.min(costBack, costBoth));
    }
}

File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: list[int]) -> int:
        n = len(nums)
        if n <= 2:
            return n

        # Find indices of min and max elements
        min_idx = nums.index(min(nums))
        max_idx = nums.index(max(nums))

        # Determine leftmost and rightmost targets
        left = min(min_idx, max_idx)
        right = max(min_idx, max_idx)

        # 1. Delete both from front
        cost_front = right + 1

        # 2. Delete both from back
        cost_back = n - left

        # 3. Delete one from front and one from back
        cost_both = (left + 1) + (n - right)

        return min(cost_front, cost_back, cost_both)